test/disable-goosig-test.js

```diff
@@ -0,0 +1,225 @@
+        assert(await node.chain.add(block));
+      }
+
+      assert.strictEqual(node.chain.height - 1, node.network.goosigStop);
+      await assert.rejects(node.mempool.addAirdrop(proof),
+        {reason: 'bad-airdrop-disabled'});
+    });
+
+    it('should reject GooSig based airdrop in block', async () => {
+      const tip = node.chain.tip;
+      const version = await node.chain.computeBlockVersion(tip);
+      const mtp = await node.chain.getMedianTime(tip);
+      const time = Math.max(node.network.now(), mtp + 1);
+      const target = await node.chain.getTarget(time, tip);
+      const root = node.chain.db.treeRoot();
+
+      const template = new BlockTemplate({
+        prevBlock: tip.hash,
+        treeRoot: root,
+        reservedRoot: consensus.ZERO_HASH,
+        height: tip.height + 1,
+        version: version,
+        time: time,
+        bits: target,
+        mtp: mtp
+      });
+
+      template.addAirdrop(proof);
+      template.refresh();
+
+      let block = template.toBlock(), nonce = 0;
+      while (!block.verifyPOW()) {
+        const proof = template.getProof(nonce++, time, consensus.ZERO_NONCE, consensus.ZERO_HASH);
+        block = template.commit(proof);
+      }
+
+      await assert.rejects(node.chain.add(block),
+        {reason: 'bad-airdrop-disabled'});
+    });
+  });
+
+  describe('Clear Mempool', () => {
+    let node;
+
+    before(async () => {
+      node = newNode();
+      await node.open();
+      await node.mempool.addAirdrop(proof);
+    });
+
+    after(async () => {
+      await node.close();
+    });
+
+    it('should remove GooSig based airdrop', async () => {
+      assert.strictEqual(node.mempool.airdrops.size, 1);
+      assert(node.mempool.has(proof.hash()));
+
+      while (node.chain.height < node.network.goosigStop) {
+        const block = await node.miner.mineBlock();
+        assert(await node.chain.add(block));
+      }
+
+      assert.strictEqual(node.mempool.has(proof.hash()), false);
+      assert.strictEqual(node.mempool.airdrops.size, 0);
+    });
+  });
+
+  describe('SPV', () => {
+    let node;
+
+    before(async () => {
+      node = newNode('spv');
+      await node.open();
+    });
+
+    after(async () => {
+      await node.close();
+    });
+
+    it('should not send airdrop after', async () => {
+      const genesis = await node.chain.getEntryByHeight(0);
+
+      let err = null;
+      node.once('error', (error) => {
+        err = error;
+      });
+
+      let prev = genesis.hash;
+      while (node.chain.height < node.network.goosigStop) {
+        const block = mockBlock(prev);
+        await node.chain.add(block, VERIFY_NONE);
+        prev = block.hash();
+      }
+
+      await node.sendAirdrop(proof);
+
+      assert(err);
+      assert.strictEqual(err.message, 'GooSig disabled.');
+    });
+  });
+});
+
+/**
+ * Create a new FullNode with GooSig
+ * disabled at a specific height.
+ */
+
+function newNode(type = 'full') {
```

Comment:
Also, how do you feel about a name change to `createNode` and moving these helpers to the top of the file? It would keep the style consistent with other test files.